### DISCUSSION TITLE
Robert Balicki - Isograph

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#271] [Announcements]
- https://twitter.com/isographlabs
- https://www.youtube.com/watch?v=SkxAOgkaiSw
- https://isograph.dev/blog/2024/02/15/isograph-0.1.0/
- https://twitter.com/StatisticsFTW
- https://www.robertbalicki.com/

### Relay

- [x] Your profiles says you were on the react data team.... what does that mean?
- [x] Relay seems to work really well for Facebook. Outside of that it hasn't really caught on, why is that?

### Isograph

- [x] What is Isograph and how is it different than other similar tools?
- [x] In 2019, there was a talk at F8 about how FB uses relay to fetch both data and code. Was that an inspiration for Isograph?
- [x] How does Isograph improve on the devx of relay?
- [x] How does it help me with data network waterfalls? Was it hard to get right?
- [x] What kind of tooling does Isograph/client fields open up?
- [x] What are some benefits of storing the react components in graphql?
- [x] For large apps does this help with perf? It seems like more suspense could take away from UX?
- [x] In what ways does the library reduce boilerplate?
- [x] What does the compiler do?
- [x] Does it work with typescript? How does it interact with the babel plugin?
- [x] How does data masking help with working in large teams?
- [x] Are you using this in production at Pinterest?

### Future

- [x] Does RSC make sense with Isograph?
- [x] What's next to get this production ready?
- [ ] What do you think the future of data in react is? Will we ever find the "one true solution"?

**Comment by hipstersmoothie** [comment]
Tooltips: 

https://github.com/facebook/react-strict-dom
https://www.npmjs.com/package/safetest

**Comment by rbalicki2** [comment]
* https://kinesis-ergo.com/shop/advantage2/
* https://folivora.ai/
* https://karabiner-elements.pqrs.org/

**Comment by just-be-dev** [comment]
Tooltips:
- https://josh-project.github.io/josh/intro.html
- https://nodedojo.gumroad.com/l/dojobin
